### PR TITLE
handle crashes with ILM expiry changes

### LINKED ISSUE #18589: segfaults with site replication
## NOTE

Since the update to version 2023-12-02T10:51:33Z, some minio nodes are crashing
We have 2 clusters with site replication, this appends one one node on each side.

```
2023-12-04_19:29:10.01902 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2387956]
2023-12-04_19:29:10.01903
2023-12-04_19:29:10.01904 goroutine 370653 [running]:
2023-12-04_19:29:10.01909 github.com/minio/minio/cmd.(*SiteReplicationSys).SiteReplicationMetaInfo(0x609f540, {0x49f14d0, 0xc0196a8be0}, {0x4a1d4d8?, 0xc0019ea000}, {0x0, 0x0, 0x0, 0x0, 0x0, ...})
2023-12-04_19:29:10.01910 	github.com/minio/minio/cmd/site-replication.go:3734 +0x3996
2023-12-04_19:29:10.01913 github.com/minio/minio/cmd.(*SiteReplicationSys).siteReplicationStatus.func1()
2023-12-04_19:29:10.01913 	github.com/minio/minio/cmd/site-replication.go:2729 +0x9f
2023-12-04_19:29:10.01916 github.com/minio/minio/cmd.(*SiteReplicationSys).concDo.func1(0x1)
2023-12-04_19:29:10.01919 	github.com/minio/minio/cmd/site-replication.go:2222 +0x135
2023-12-04_19:29:10.01922 created by github.com/minio/minio/cmd.(*SiteReplicationSys).concDo in goroutine 256752
2023-12-04_19:29:10.01922 	github.com/minio/minio/cmd/site-replication.go:2218 +0x25f
```

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
handle crashes with ILM expiry changes

## Motivation and Context
there is a typo that causes site replication to crash

fixes #18589 

## How to test this PR?
As per #18589 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by poornas — APPROVED** [review]

File: cmd/site-replication.go [hunk 1]
```diff
@@ -3723,7 +3723,7 @@ func (c *SiteReplicationSys) SiteReplicationMetaInfo(ctx context.Context, objAPI
 				return info, errSRBackendIssue(err)
 			}
 
-			if meta.lifecycleConfig != nil && meta.lifecycleConfig.HasExpiry() {
+			if meta.lifecycleConfig != nil && meta.lifecycleConfig.HasExpiry() && meta.lifecycleConfig.ExpiryUpdatedAt != nil {
```
Comment (krisis) [inline cmd/site-replication.go]: IIUC, we should copy over ILM expiry rules even when `ExpiryUpdatedAt` is nil. When users upgrade to a MinIO version with this feature (and fix), they will have ILM policies with expiry rules with no `ExpiryUpdatedAt`(i.e `nil`). We should add the `!= nil` check few lines below like so,
```go
var updatedAt time.Time
if meta.lifecycleConfig.ExpiryUpdatedAt != nil {
    updatedAt = *meta.lifecycleConfig.ExpiryUpdatedAt
}
allRules[rule.ID] = madmin.ILMExpiryRule{ILMRule: string(ruleData), Bucket: bucket, UpdatedAt: updatedAt}
```

Comment (harshavardhana) [reply cmd/site-replication.go]: is that true? @krisis isn't it expected that we send ILM rules only when we have this value correct

**Review by jiuker — APPROVED** [review]

File: cmd/site-replication.go (hunk 1 above)
Comment (krisis) [reply cmd/site-replication.go]: @shtripat will know best. ILM policies created before the MinIO version which introduced `ExpiryUpdatedAt` field will have this field to be `nil`. Unless this is handled specially during server initialization, we would have to handle this when we 'heal' ILM expiry rules between sites in site-replication.

**Comment by harshavardhana** [comment]
@shtripat does this fix look good to you?

**Review by shtripat — APPROVED** [review]
lgtm

**Pull request merged**